src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -745,10 +745,10 @@ - (void)setDelegate:(id<MTRDeviceDelegate>)delegate queue:(dispatch_queue_t)queu
         if ([self _deviceUsesThread]) {
             [self _scheduleSubscriptionPoolWork:^{
                 std::lock_guard lock(self->_lock);
-                [self _setupSubscription];
+                [self _setupSubscriptionWithReason:@"delegate is set and subscription is scheduled"];
             } inNanoseconds:0 description:@"MTRDevice setDelegate first subscription"];
         } else {
-            [self _setupSubscription];
+            [self _setupSubscriptionWithReason:@"delegate is set and subscription is needed"];
```

Comment:
on one hand we could make these constants

on the other hand we could we could make it more clear that the method is just using the string for logging  (`_setupSubscriptionWithLoggedReason:`)

Reply:
Hmm, I like the second version of naming the methods better. Right now I'm using the same convention as `_triggerResubscribeWithReason:` and so I think we can stick with this for now, and rename everything in a separate patch.